cheri_compressed_cap_common.h

```diff
@@ -346,12 +358,7 @@ static inline bool _cc_N(compute_base_top)(_cc_bounds_bits bounds, _cc_addr_t cu
         *top_out = _CC_N(MAX_TOP);
         return valid;
     }
-
-    // Remove flags bits
-    cursor = cursor & _CC_CURSOR_MASK;
-    // Sign extend
-    if (cursor & ((_CC_CURSOR_MASK >> 1) + 1))
-        cursor |= ~_CC_CURSOR_MASK;
+    cursor = _cc_N(morello_cap_bounds_address)(cursor);
```

Comment:
If _CC_CURSOR_MASK is ~0 on !cc128m then you can just do this unconditionally, with it becoming a slightly stupid identity

Reply:
Done, thanks for the suggestion.